test/test_theme.rb

```diff
@@ -0,0 +1,55 @@
+require 'helper'
+
+class TestTheme < JekyllUnitTest
+  def setup
+    @theme = Theme.new('test-theme')
+    @expected_root = File.expand_path "./fixtures/test-theme", File.dirname(__FILE__)
+  end
+
+  context "initializing" do
+    should "normalize the theme name" do
+      theme = Theme.new(' Test-Theme ')
+      assert_equal "test-theme", theme.name
+    end
+
+    should "know the theme root" do
+      assert_equal @expected_root, @theme.root
+    end
+
+    should "know the theme version" do
+      assert_equal Gem::Version.new("0.1.0"), @theme.version
+    end
+
+    should "raise an error for invalid themes" do
+      assert_raises Jekyll::Errors::MissingDependencyException do
+        Theme.new("foo").version
+      end
+    end
+  end
+
+  context "path generation" do
+    [:layouts, :includes, :sass].each do |folder|
+      should "know the #{folder} path" do
+        expected = File.expand_path("_#{folder}", @expected_root)
+        assert_equal expected, @theme.public_send("#{folder}_path")
+      end
+    end
+
+    should "generate folder paths" do
+      expected = File.expand_path("./_sass", @expected_root)
+      assert_equal expected, @theme.send(:path_for, :sass)
+    end
+
+    should "not allow paths outside of the theme root" do
+      assert_equal nil, @theme.send(:path_for, "../../source")
```

Comment:
> how about resolving the path relative to the site source

Do you mean relative to the theme directory?

> should `path_for "folder/folder"` give `_folder/folder` or `folder/_folder`?

That's why I'm interested in changing the method. `path_for "folder/folder"` should resolve the path `theme_dir/folder/folder`. The argument to `path_for` should be responsible for placing the `_` so we don't have to worry about this.